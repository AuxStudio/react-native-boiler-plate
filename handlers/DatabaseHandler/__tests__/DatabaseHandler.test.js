import React from 'react';
import renderer from 'react-test-renderer';

import { DatabaseHandler } from '../';

describe('DatabaseHandler', () => {
  let spy;
  const dispatch = jest.fn();

  it('renders with all props', () => {
    expect(
      renderer.create(<DatabaseHandler dispatch={jest.fn()} authenticated />),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<DatabaseHandler dispatch={jest.fn()} />)).toMatchSnapshot();
  });

  it('calls listenForData on componentDidMount if authenticated prop is supplied', () => {
    spy = jest.spyOn(DatabaseHandler.prototype, 'listenForData');

    renderer.create(<DatabaseHandler dispatch={dispatch} authenticated />);

    expect(spy).toHaveBeenCalled();
  });

  it('does not call listenForData on componentDidMount if authenticated prop is not supplied', () => {
    spy = jest.spyOn(DatabaseHandler.prototype, 'listenForData');

    renderer.create(<DatabaseHandler dispatch={dispatch} />);

    expect(spy).not.toHaveBeenCalled();
  });

  afterEach(() => {
    if (spy) {
      spy.mockReset();
    }
  });
});
