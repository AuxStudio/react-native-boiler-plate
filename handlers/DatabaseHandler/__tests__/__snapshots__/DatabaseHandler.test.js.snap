// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatabaseHandler renders with all props 1`] = `null`;

exports[`DatabaseHandler renders with minimum required props 1`] = `null`;
